### ISSUE TITLE
[datepicker] [carbon-component-react]: Carbon component react datepicker component does ot work in android 12 when using talkback voice reader

### ISSUE THREAD

**Issue by shivaakumar** [carbon-design-system/carbon#12619]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

Android 12

### Package version

7.34.0

### React version

16.14.0

### Automated testing tool and ruleset

NA

### Assistive technology

Talkback in android

### Description

Carbon component react datepicker component does ot work in android 12 when using talkback voice reader

### Reproduction/example

Open chrome browser in andriod 12 and try accessing the datepicker component after turning on talkback voice reader. The datepicker does not show the date popup

### Steps to reproduce

1) Open chrome browser in andriod 12 
2) Enable talkback voice reader fucntionality
3) Try accessing the datepicker component with voice reader.
4) The datepicker popup does not show up.

Note- It works in Android 11

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
@shivaakumar can you try updating to the latest version of Carbon (`7.59.5`) and see if this is still an issue? `7.34.0` is quite an [old version](https://www.npmjs.com/package/carbon-components-react/v/7.34.0) of the components, and I'm wondering if this issue may have been resolved already.

**Comment by shivaakumar** [comment]
Thanks for a quick reply. I tried updating to 7.59.5 version but i still see the issue in chrome browser of android 12. 

FYI if this may help you, when the focus is on datepicker, the voice reader reads as "disabled".

**Comment by tw15egan** [comment]
Interesting 🤔  We utilize Flatpickr under the hood for our Datepicker. Can you check their [example website](https://flatpickr.js.org/examples/) to see if the issues also exist? I cannot because I do not have access to an Android device.  
 If there is an issue there as well, we'd need to open an issue on their end to get this resolved. Thanks!

**Comment by shivaakumar** [comment]
Its actually because of property allowInput to Datepicker component which is causing the issue. This adds readonly property to input field. I suspect that may be the reason it does not work after enabling talkback in android. 

Any suggestions to fix this issue or is this a known issue ?

**Comment by shivaakumar** [comment]
Its actually because of property allowInput to Datepicker component which is causing the issue. This adds readonly property to input field. I suspect that may be the reason it does not work after enabling talkback in android.

Any suggestions to fix this issue or is this a known issue ?
